resources/lib/WonderfulSubsBrowser.py

```diff
@@ -97,14 +99,14 @@ def _handle_watchlist_paging(self, results, base_url, page):
         name = "Next Page (%d/%d)" % (next_page, total_pages)
         return [utils.allocate_item(name, base_url % next_page, True, None)]
 
-    def _json_request(self, url, data):
-        response = json.loads(self._get_request(url, data))
-        if response["status"] != 200:
-            raise Exception("Request %s returned with error code %d" % (url,
-                                                                        response["status"]))
+    def _json_request(self, url, data, reauth=True):
+        response = self._get_request(url, data)
 
-        return response["json"]
+        if self._response_forbidden(response) and reauth:
```

Comment:
Yeah this response foribidden is part of this class, not the base..

Reply:
And you can still get non-200 which is not foribbdden

Reply:
This part only deals with actual forbidden responses. Everything else should still be an error. It just makes it easy to (re)login, if credentials are stored in the settings, but no token exists or it needs refreshing.

Reply:
Yes but it should be with on the same class

Reply:
Yeah, now I got it. Makes perfect sense! Thanks!